math/fibonacci_matrix_exponentiation.cpp

```diff
@@ -0,0 +1,138 @@
+   /**
+ * @file fibonacci_matrix_exponentiation.cpp
+ * @brief Computes N^th Fibonacci number given as
+ * input argument
+ *
+ * Takes O(logn) time to compute nth Fibonacci number
+ * 
+ *
+ * \author [villayatali123](https://github.com/villayatali123)
+ * \author [unknown author]()
+ * @see fibonacci.cpp, fibonacci_fast.cpp, string_fibonacci.cpp, fibonacci_large.cpp
+ */
+/* @details
+ *Begin 
+ * Take two 2 dimensional vector
+ * Create a function and Perform matrix multiplication
+ * Ceate another function to find out power of matrix
+ * Create a function then to find out the Fibonacci number
+ * Multiply(arr1[2][2], arr2[2][2], int size)
+ * Take 4 variables a, b, c, d
+ * a = arr1[0][0] * arr2[0][0] + arr1[0][1] * arr2[1][0]
```

Comment:
```suggestion
 * ```
 * a = arr1[0][0] * arr2[0][0] + arr1[0][1] * arr2[1][0]
```